### PR TITLE
Upgrade scala and finagle dependencies to latest versions

### PR DESCRIPTION
**Pull request by ddelnano**
With the netty (https://github.com/netty/netty/pull/12438) and netty-tcnative (https://github.com/netty/netty-tcnative/pull/731) PRs required for #407  being actively reviewed, we will need to update finagle very soon. This PR gets us to the latest version so that the subsequent upgrade will be a smaller change.

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -138,14 +138,20 @@ load("//bazel:thrift.bzl", "thrift_deps")
...
+# https://github.com/bazelbuild/rules_scala/pull/847 for more details.
 bind(
     name = "io_bazel_rules_scala/dependency/thrift/scrooge_core",
     actual = "//src/stirling/source_connectors/socket_tracer/testing/containers/thriftmux:scrooge_jars",
 )
 
+bind(
+    name = "io_bazel_rules_scala/dependency/thrift/scrooge_generator",
+    actual = "//src/stirling/source_connectors/socket_tracer/testing/containers/thriftmux:scrooge_jars",
+)
```
Comment (ddelnano) [inline WORKSPACE:153]: This was missed when we first got this working, but it went unnoticed until there was a breaking change made to scrooge generator (which happened in 21.9.0). Without this `bind` statement the rules_scala defaults to an older version and causes the `src/stirling/source_connectors/socket_tracer/testing/containers/thriftmux:server_image` target to fail to build due to a compilation error.

File: src/stirling/source_connectors/socket_tracer/testing/containers/thriftmux/src/main/scala/server/Server.scala [hunk 2]
```diff
@@ -20,7 +20,7 @@ object Server {
       new File("/etc/ssl/server.key"),
     )
     val sslConfig = SslServerConfiguration(
-      keyCredentials,
+      keyCredentials=keyCredentials,
```
Comment (ddelnano) [inline src/stirling/source_connectors/socket_tracer/testing/containers/thriftmux/src/main/scala/server/Server.scala:23]: The order of these positional arguments changed. Switching to named parameters should hopefully future proof this.

**Comment by ddelnano** [comment]
@oazizi000 can you give this a review when you have the chance?

**Review by oazizi000 — APPROVED** [review]

**Comment by oazizi000** [comment]
@ddelnano Is this dependent on https://github.com/pixie-io/pixie/pull/446?

**Comment by ddelnano** [comment]
Nope, this change is good in its own right. I will eventually need to rebase #446 because it will need this change (installing latest finagle once my netty changes are released).

**Pull request closed without merging**